MC/bin/o2dpg_sim_workflow_anchored.py

```diff
@@ -215,16 +222,26 @@ def retrieve_MinBias_CTPScaler_Rate(ccdbreader, timestamp, run_number, finaltime
     ts, ctpscaler = ccdbreader.fetch(path, "o2::ctp::CTPRunScalers", timestamp = timestamp)
     if ctpscaler != None:
       ctpscaler.convertRawToO2()
-      rate = ctpscaler.getRateGivenT(finaltime,0,0)  # the simple FT0 rate from the counters
-      # print("Global rate " + str(rate.first) + " local rate " + str(rate.second))
-
-      # now get the bunch filling object which is part of GRPLHCIF and calculate
-      # true rate (input from Chiara Zampolli)
-      ts, grplhcif = ccdbreader.fetch("GLO/Config/GRPLHCIF", "o2::parameters::GRPLHCIFData", timestamp = timestamp)
-      coll_bunches = grplhcif.getBunchFilling().getNBunches()
-      mu = - math.log(1. - rate.first / 11245 / coll_bunches) / ft0_eff
-      finalRate = coll_bunches * mu * 11245
-      return finalRate
+      # this is the default for pp
+      ctpclass = 0 # <---- we take the scaler for FT0
+      ctptype = 0
+      # this is the default for PbPb
+      if ColSystem == "PbPb":
+        ctpclass = 25  # <--- we take scalers for ZDC
+        ctptype = 7
+      print("Fetching rate with class " + str(ctpclass) + " type " + str(ctptype))
+      rate = ctpscaler.getRateGivenT(finaltime, ctpclass, ctptype)
+      #if ColSystem == "PbPb":
+      #  rate.first = rate.first / 28.
+      #  rate.second = rate.second / 28.
+
+      print("Global rate " + str(rate.first) + " local rate " + str(rate.second))
+      if rate.first >= 0:
+        # calculate true rate (input from Chiara Zampolli) using number of bunches
+        coll_bunches = NBunches
+        mu = - math.log(1. - rate.second / 11245 / coll_bunches) / ft0_eff
+        finalRate = coll_bunches * mu * 11245
```

Comment:
@chiarazampolli @noferini : This is better than before. Class 25 seems to get some numbers for PbPb runs. However, we need to carefully review the other numbers and formulas. For the moment the interaction rate seems larger than typical PbPb rates.

Reply:
Hello @sawenzel , 
Can you give us some examples?

Reply:
run 544508 : Interaction rate 262397

Reply:
We can take a look. What are you comparing this to?
The IRs in the spreadsheet are integrated over some time (depending on what you use), so some difference is expected since PbPb is not levelled, but it should be reasonable. Unless you are comparing to smth else than the spreadsheet. 
Note that the get-run-parameter (or similar) in O2 should be also updated for PbPb.

Reply:
I guess that the rate is obtained assuming a ft0_eff=4 (corr factor for ft0).
If for ZDC we expect a scale factor 28 (do I remember correctly?) you should find an IR 7 time larger (it seems the case).
It could be that we need only to adjust the trigger efficiency in our scripts

Reply:
@noferini : Might well be. I would like to delegate this to you or someone else.

Reply:
I added a comment to the line you have to change

Reply:
The factor 28 is explained here by Roman:

https://alice.its.cern.ch/jira/browse/O2-4315?focusedId=308708&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-308708